sts3x/Makefile

```diff
@@ -1,43 +1,19 @@
-sensirion_common_dir := ../embedded-common
-sts_common_dir := ../sts-common
-sw_i2c_dir := ${sensirion_common_dir}/sw_i2c
-hw_i2c_dir := ${sensirion_common_dir}/hw_i2c
+UNAME_S := $(shell uname -s)
+ifeq ($(UNAME_S), Linux)
+	# On Linux, use the sample implementation instead of the empty stubs
+	# Only one of those is used depending on CONFIG_I2C_TYPE (default is hw_i2c)
+	hw_i2c_impl_src ?= ${sensirion_common_dir}/hw_i2c/sample-implementations/linux_user_space/sensirion_hw_i2c_implementation.c
+	sw_i2c_impl_src ?= ${sensirion_common_dir}/sw_i2c/sample-implementations/linux_user_space/sensirion_sw_i2c_implementation.c
+endif
 
-CFLAGS := -Os -fstrict-aliasing -Wstrict-aliasing=1 -Wall -Wsign-conversion \
-          -I. -I${sensirion_common_dir} -I${sts_common_dir}
-
-sensirion_common_objects := sensirion_common.o
-sts_common_objects := sts_git_version.o
-sts3x_binaries := sts3x_example_usage_sw_i2c sts3x_example_usage_hw_i2c
-sts_binaries += ${sts3x_binaries}
-
-sw_objects := ${sensirion_common_dir}/sw_i2c/sensirion_sw_i2c.o ${sensirion_common_dir}/sw_i2c/sensirion_sw_i2c_implementation.o
-hw_objects := ${sensirion_common_dir}/hw_i2c/sensirion_hw_i2c_implementation.o
-all_objects := ${sensirion_common_objects} ${sts_common_objects} ${hw_objects} ${sw_objects} sts3x.o
+include config.inc
 
 .PHONY: all
 
-all: ${sts_binaries}
-
-sts_git_version.o: ${sts_common_dir}/sts_git_version.c
-	$(CC) $(CFLAGS) -c -o $@ $^
-sensirion_common.o: ${sensirion_common_dir}/sensirion_common.c
-	$(CC) $(CFLAGS) -c -o $@ $^
-sensirion_sw_i2c_implementation.o: ${sw_i2c_dir}/sensirion_sw_i2c_implementation.c
-	$(CC) $(CFLAGS) -c -o $@ $^
-sensirion_hw_i2c_implementation.o: ${hw_i2c_dir}/sensirion_hw_i2c_implementation.c
-	$(CC) $(CFLAGS) -c -o $@ $^
-
-sensirion_sw_i2c.o: ${sw_i2c_dir}/sensirion_sw_i2c.c
-	$(CC) -I${sw_i2c_dir} $(CFLAGS) -c -o $@ $^
-
-sts3x.o: ${sensirion_common_dir}/sensirion_i2c.h ${sts_common_dir}/sts_git_version.c sts3x.h sts3x.c
-
-sts3x_example_usage_sw_i2c: ${sensirion_common_objects} ${sts_common_objects} ${sw_objects} sts3x.o
-	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ $(LOADLIBES) $(LDLIBS) sts3x_example_usage.c
+all: sts3x_example_usage_${CONFIG_I2C_TYPE}
```

Comment:
@rnestler what do you think about removing the variable from the target.. I expect people to only build one target anyway. I'd also change it to always rebuild since it won't be able to pick up changes stemming from `config.inc`, env var etc.

```
sts3x_example_usage:
    $(CC) $(CFLAGS) -o $@ ${sts3x_sources} ${${CONFIG_I2C_TYPE}_sources} ${sts3x_dir}/sts3x_example_usage.c
```